lib/hanami/cli/generators/app/ruby_file_writer.rb

```diff
@@ -0,0 +1,149 @@
+# frozen_string_literal: true
+
+require "erb"
+require "dry/files"
+require_relative "../constants"
+require_relative "../../errors"
+
+module Hanami
+  module CLI
+    module Generators
+      module App
+        # @since 2.2.0
+        # @api private
+        class RubyFileWriter
+          # @since 2.2.0
+          # @api private
+          def initialize(
+            fs:,
+            inflector:,
+            app_namespace:,
+            key:,
+            slice:,
+            relative_parent_class:,
+            extra_namespace: nil,
+            body: []
```

Comment:
Like you mentioned in the description, this is an internal class only so we can tolerate some slightly less ergonomic API design, but I think a refactor to allow a FileWriter instance to be initialised once and then used to write multiple files might be nice.

So this would probably pull the `fs:` and `inflector:` params out into `#initialize`, with the rest going into some method that writes an actual file.

I think this'd be worth playing with sometime, but it's not blocking and I'm definitely keen to get our full range of CLI commands ready for beta1 on Tuesday. So happy if this comes in a follow-up change of some kind :)

Reply:
Made a note for myself to explore this later